### PR TITLE
feat: integrate with mapcolonies config manager

### PR DESCRIPTION
**Pull request by ronenkapelian**
Integrate service with config managment stack including dedicated schema for opa-la

| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: packages/auth-bundler/package.json [hunk 1]
```diff
@@ -47,7 +45,9 @@
     "jest-extended": "^3.2.4",
     "rimraf": "^4.4.0",
     "ts-node": "^10.9.1",
-    "typescript": "^4.9.5"
+    "typescript": "^4.9.5",
+    "@map-colonies/schemas": "file:/home/ronenk1/dev/infra/schemas",
```
[the remark this answers is not in the document]
Comment (ronenkapelian) [reply packages/auth-bundler/package.json:49]: good idea

File: packages/auth-bundler/tests/opa.spec.ts [hunk 2]
```diff
@@ -63,14 +62,14 @@ describe('opa.ts', function () {
 
   describe('#validateBinaryExistCommand', function () {
     it('should return true if the binary exists', async function () {
-      jest.spyOn(execa, 'execa').mockResolvedValue({ failed: false } as ExecaReturnValue);
+      jest.spyOn(execa, 'execa').mockResolvedValue({ failed: false } as Awaited<ReturnType<(typeof execa)['execa']>>);
```
[the remark this answers is not in the document]
Comment (ronenkapelian) [reply packages/auth-bundler/tests/opa.spec.ts:65]: good comment :) changed.

File: packages/auth-core/package.json [hunk 3]
```diff
@@ -45,7 +43,10 @@
     "copyfiles": "^2.4.1",
     "rimraf": "^4.4.0",
     "ts-node": "^10.9.1",
-    "typescript": "^4.9.5"
+    "typescript": "^4.9.5",
+    "@map-colonies/schemas": "file:/home/ronenk1/dev/infra/schemas",
+    "@map-colonies/config": "^1.1.0"
```
[the remark this answers is not in the document]
Comment (ronenkapelian) [reply packages/auth-core/package.json:48]: done

File: packages/auth-core/src/config.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ * @returns A Promise that resolves when the configuration is successfully initialized.
+ */
+async function initConfig(offlineMode?: boolean): Promise<void> {
+  configInstance = await config({
+    configName: 'auth-core',
+    configServerUrl: 'http://localhost:8080',
+    schema: commonDbFullV1,
+    version: 'latest',
+    offlineMode: offlineMode,
+  });
```
[the remark this answers is not in the document]
Comment (ronenkapelian) [reply packages/auth-core/src/config.ts:20]: done

File: packages/auth-core/src/db/types/interfaces.ts [hunk 5]
```diff
@@ -3,10 +3,22 @@ import { PostgresConnectionOptions } from 'typeorm/driver/postgres/PostgresConne
...
+        enabled: true;
+      }
+    | {
+        [x: string]: unknown;
+        ca?: string | undefined;
+        cert?: string | undefined;
+        key?: string | undefined;
+        enabled: false;
+      };
 } & PostgresConnectionOptions;
```
[the remark this answers is not in the document]
Comment (ronenkapelian) [reply packages/auth-core/src/db/types/interfaces.ts:24]: changed, now it using part of schema original entity

File: packages/auth-core/src/instrumentation.mts
[the remark this answers is not in the document]
Comment (ronenkapelian) [reply packages/auth-core/src/instrumentation.mts:1]: image

its already include migrations script, so on such use case it is needed

File: packages/auth-cron/src/config.ts [hunk 6]
```diff
@@ -1,139 +1,30 @@
...
+ * This should only be called from the instrumentation file.
+ * @returns A Promise that resolves when the configuration is successfully initialized.
+ */
+async function initConfig(offlineMode?: boolean): Promise<void> {
+  configInstance = await config({
+    configName: 'auth-cron',
+    configServerUrl: 'http://localhost:8080',
+    schema: infraOpalaCronV1,
+    version: 'latest',
+    offlineMode: offlineMode,
```
[the remark this answers is not in the document]
Comment (ronenkapelian) [reply packages/auth-cron/src/config.ts:19]: done

File: packages/auth-core/src/instrumentation.mts
[the remark this answers is not in the document]
Comment (CptSchnitz) [reply packages/auth-core/src/instrumentation.mts:1]: reminding you, it should not be part of src

File: packages/auth-cron/src/config.ts [hunk 7]
```diff
@@ -12,10 +12,8 @@ let configInstance: ConfigType | undefined;
  */
 async function initConfig(offlineMode?: boolean): Promise<void> {
   configInstance = await config({
-    configName: 'auth-cron',
     configServerUrl: 'http://localhost:8080',
```
Comment (CptSchnitz) [inline packages/auth-cron/src/config.ts:15]: can also be removed

**Review by CptSchnitz — CHANGES_REQUESTED** [review]
you can also release a new version of schemas and add it to the PR

File: packages/auth-bundler/config/custom-environment-variables.json
[the remark this answers is not in the document]
Comment (ronenkapelian) [reply packages/auth-bundler/config/custom-environment-variables.json:1]: it seems you are right, there is no need for this module. i will delete

File: packages/auth-cron/src/config.ts (hunk 7 above)
Comment (ronenkapelian) [reply packages/auth-cron/src/config.ts:15]: removed

File: packages/auth-core/src/instrumentation.mts
[the remark this answers is not in the document]
Comment (ronenkapelian) [reply packages/auth-core/src/instrumentation.mts:1]: removed

File: packages/auth-cron/tests/validators.spec.ts [hunk 8]
```diff
@@ -1,96 +1,23 @@
...
-              bucket: 'bundles',
-              key: 'np.tar.gz',
-              endpoint: 'http://localhost:9000',
-            },
-          },
-        },
-      };
-
-      expect(() => validateConfigSchema(schema)).toThrow();
-    });
```
[the remark this answers is not in the document]
Comment (ronenkapelian) [reply packages/auth-cron/tests/validators.spec.ts:87]: you name it

File: packages/auth-bundler/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (ronenkapelian) [reply packages/auth-bundler/package.json]: updated to v1.1.0 of schemas

**Review by netanelC — CHANGES_REQUESTED** [review]
Good Job!

File: packages/auth-bundler/tests/configurations/jest.globalSetup.ts [hunk 9]
```diff
@@ -1,23 +1,29 @@
...
+import { getConfig, initConfig } from '../helpers/config';
 
 export default async (): Promise<void> => {
   const folder = path.join(tmpdir(), 'authbundlertests');
   if (!existsSync(folder)) {
     await mkdir(folder);
   }
 
-  const dataSourceOptions = config.get<DbConfig>('db');
+  // eslint-disable-next-line @typescript-eslint/no-magic-numbers
```
[the remark this answers is not in the document]
Comment (ronenkapelian) [reply packages/auth-bundler/tests/configurations/jest.globalSetup.ts:14]: removed

File: packages/auth-manager/tests/integration/client/client.spec.ts [hunk 10]
```diff
@@ -11,11 +11,15 @@ import { getApp } from '../../../src/app';
...
 import { ClientRepository } from '../../../src/client/DAL/clientRepository';
+import { initConfig } from '../../../src/common/config';
 import { ClientRequestSender } from './helpers/requestSender';
 
 describe('client', function () {
   let requestSender: ClientRequestSender;
   let depContainer: DependencyContainer;
+  beforeAll(async function () {
+    await initConfig(true);
+  });
```
[the remark this answers is not in the document]
Comment (ronenkapelian) [reply packages/auth-manager/tests/integration/client/client.spec.ts:22]: done

File: packages/auth-manager/tests/integration/domain/domain.spec.ts [hunk 11]
```diff
@@ -8,11 +8,15 @@ import 'jest-openapi';
...
 import { SERVICES } from '../../../src/common/constants';
+import { initConfig } from '../../../src/common/config';
 import { DomainRequestSender } from './helpers/requestSender';
 
 describe('domain', function () {
   let requestSender: DomainRequestSender;
   let depContainer: DependencyContainer;
+  beforeAll(async function () {
+    await initConfig(true);
+  });
```
[the remark this answers is not in the document]
Comment (ronenkapelian) [reply packages/auth-manager/tests/integration/domain/domain.spec.ts:19]: done

File: packages/auth-manager/config/default.json [hunk 12]
```diff
@@ -9,10 +9,14 @@
     "logger": {
       "level": "info",
       "prettyPrint": false
-    }
+    },
+    "tracing": {
+      "isEnabled": false
+    },
+    "shared": {}
```
[the remark this answers is not in the document]
Comment (ronenkapelian) [reply packages/auth-manager/config/default.json:16]: image

it is part of the boilerplate schema: 
https://github.com/MapColonies/schemas/blob/master/schemas/common/boilerplate/v1.schema.json

File: packages/auth-bundler/tests/tsconfig.json [hunk 13]
```diff
@@ -0,0 +1,5 @@
+{
+    "extends": "../tsconfig.json",
+    "include": ["."],
```
[the remark this answers is not in the document]
Comment (ronenkapelian) [reply packages/auth-bundler/tests/tsconfig.json:3]: yes, it is for auto-completion of types inside this tests

File: packages/auth-manager/config/default.json (hunk 12 above)
[the remark this answers is not in the document]
Comment (netanelC) [reply packages/auth-manager/config/default.json:16]: It is not required, if you keep it as an empty object, don't put it

**Review by netanelC — COMMENTED** [review]
Fix the issues with the config package as we saw today and then this will be considered as approved to me. Wait for Ofer to final approve and merge it

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**